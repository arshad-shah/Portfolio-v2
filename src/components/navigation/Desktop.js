import React from "react";
import { Box, Typography } from "@material-ui/core";
import { Link } from "react-scroll";
import Resume from "../../assets/ArshadResume.pdf";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  menuItem: {
    marginRight: "1.5rem",
    "&:hover": {
      backgroundColor: "#fff152",
    },
    "&:active": {
      backgroundColor: "transparent",
    },
    "&:visited": {
      backgroundColor: "transparent",
    },
    "&:link": {
      backgroundColor: "none",
    },
    color: "#000000",
    textDecoration: "none",
    cursor: "pointer",
    borderRadius: "0.5rem",
    transition: "all 0.5s ease",
    padding: "1rem",
  },
}));
export default function Desktop() {
  const classes = useStyles();
  return (
    <Box aria-label="desktopMenu" data-testid="desktopMenu">
      <Link
        data-testid="aboutme"
        to="aboutme"
        spy={true}
        smooth={true}
        data-block="Aboutmemenuitemdesktop"
        color="default"
        aria-label="About Me"
        className={classes.menuItem}
      >
        <Typography variant="button">About Me</Typography>
      </Link>
      <Link
        data-testid="projects"
        to="projects"
        spy={true}
        smooth={true}
        data-block="Projectsmenuitemdesktop"
        color="default"
        aria-label="Projects"
        className={classes.menuItem}
      >
        <Typography variant="button">Projects</Typography>
      </Link>
      <a
        data-testid="resume"
        aria-label="Resume"
        data-block="Resumemenuitemdesktop"
        href={Resume}
        color="default"
        className={classes.menuItem}
      >
        <Typography variant="button">Resume</Typography>
      </a>
    </Box>
  );
}
