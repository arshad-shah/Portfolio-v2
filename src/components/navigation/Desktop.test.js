import React from "react";
import { render, screen } from "@testing-library/react";
import Desktop from "./Desktop";

describe("Desktop of the site:", () => {
  it("should render Desktop menu with 3 buttons", () => {
    render(<Desktop />);
    const menuItem = screen.getAllByLabelText("menuItem");
    expect(menuItem.length).toEqual(3);
  });
  it("should render Desktop menu with the About me button", () => {
    render(<Desktop />);
    const aboutme = screen.getByText("About Me");
    expect(aboutme).toBeInTheDocument();
  });

  it("should render Desktop menu with the Projects button", () => {
    render(<Desktop />);
    const Projects = screen.getByText("Projects");
    expect(Projects).toBeInTheDocument();
  });

  it("should render Desktop menu with the Resume button", () => {
    render(<Desktop />);
    const Resume = screen.getByText("Resume");
    expect(Resume).toBeInTheDocument();
  });

  it("should render Desktop", () => {
    const { container } = render(<Desktop />);
    expect(container).toMatchSnapshot();
  });
});
