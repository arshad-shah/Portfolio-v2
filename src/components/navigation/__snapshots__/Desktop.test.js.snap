// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop of the site: should render Desktop 1`] = `
<div>
  <div
    aria-label="desktopMenu"
  >
    <a
      aria-label="About Me"
      class="makeStyles-menuItem-4"
      color="default"
      data-block="Aboutmemenuitemdesktop"
    >
      <span
        class="MuiTypography-root MuiTypography-button"
      >
        About Me
      </span>
    </a>
    <a
      aria-label="Projects"
      class="makeStyles-menuItem-4"
      color="default"
      data-block="Projectsmenuitemdesktop"
    >
      <span
        class="MuiTypography-root MuiTypography-button"
      >
        Projects
      </span>
    </a>
    <a
      aria-label="Resume"
      class="makeStyles-menuItem-4"
      color="default"
      data-block="Resumemenuitemdesktop"
      href="ArshadResume.pdf"
    >
      <span
        class="MuiTypography-root MuiTypography-button"
      >
        Resume
      </span>
    </a>
  </div>
</div>
`;
